enhancements/okd/okd-featureset.md

```diff
@@ -0,0 +1,638 @@
+
+### User Stories
+
+As an OKD cluster administrator, I want to enable TechPreview features on my cluster so that I can test upcoming OpenShift functionality and provide community feedback, without permanently blocking my ability to upgrade. I also want to adopt new features earlier in the lifecycle compared to OCP.
+
+As an OKD developer, I want to help users adopt new features in the community distribution so that I can validate functionality before it reaches OpenShift customers.
+
+As a feature developer, I want OKD clusters to adopt new capabilities so that I can gather early feedback from the community before promoting features to GA.
+
+As an OpenShift engineer, I want to prevent the OKD feature set from being enabled on OpenShift clusters so that we maintain clear boundaries between community and supported distributions.
+
+### Goals
+
+The goals of this proposal are:
+
+* Introduce a new "OKD" feature set in the OpenShift API
+* Enable the OKD feature set by default on all OKD clusters
+* Allow OKD clusters to enable features from TechPreview while supporting upgrades
+* Prevent the OKD feature set from being enabled on OpenShift Container Platform clusters
+* Ensure proper validation and immutability of the OKD feature set once enabled
+* Generate appropriate CRD manifests for all OpenShift API resources with the OKD feature set
+
+### Non-Goals
+
+* Modifying the behavior of existing feature sets (Default, TechPreviewNoUpgrade, DevPreviewNoUpgrade, CustomNoUpgrade)
+* Changing how feature gates work in OpenShift Container Platform
+* Providing a mechanism to disable the OKD feature set once enabled on an OKD cluster
+* Supporting the OKD feature set on OpenShift Container Platform installations
+
+## Proposal
+
+This proposal introduces a new "OKD" feature set to the OpenShift API configuration types. The feature set will:
+
+1. Be added as a new constant in `config/v1/types_feature.go`
+2. Include validation rules that prevent changing the feature set once enabled
+3. Block enablement on OpenShift clusters with appropriate error messages
+4. Generate CRD manifests for all API versions supporting the OKD variant
+5. Be enabled by default on OKD cluster installations
+6. Be able to transition from OKD to TechPreviewNoUpgrade or DevPreviewNoUpgrade
+
+### Workflow Description
+
+**OKD cluster administrator** is a user deploying and managing an OKD cluster.
+
+**OpenShift cluster administrator** is a user deploying and managing an OpenShift Container Platform cluster.
+
+**Feature developer** is an engineer developing new features for OpenShift.
+
+#### Scenario 1: Installing a new OKD cluster
+
+1. OKD cluster administrator initiates a cluster installation using `openshift-install` built for OKD
+2. The installer automatically sets the feature set to "OKD" in the cluster configuration
```

Comment:
we need to change this flow here - the CVO will be the one responsible for setting the featuregate to OKD as suggested [here](https://github.com/openshift/api/pull/2451#discussion_r2487674862)